test/concepts/test_concepts_sender.cpp

```diff
@@ -77,11 +86,13 @@ TEST_CASE("can query completion signatures for a typed sender that sends nothing
   check_val_types<type_array<type_array<>>>(my_sender0{});
   check_err_types<type_array<std::exception_ptr>>(my_sender0{});
   check_sends_stopped<true>(my_sender0{});
+  REQUIRE(ex::sender_of<my_sender0, ex::set_value_t()>);
 }
 TEST_CASE("can query completion signatures for a typed sender that sends int", "[concepts][sender]") {
   check_val_types<type_array<type_array<int>>>(my_sender_int{});
   check_err_types<type_array<std::exception_ptr>>(my_sender_int{});
   check_sends_stopped<true>(my_sender_int{});
+  REQUIRE(ex::sender_of<my_sender_int, ex::set_value_t(int)>);
```

Comment:
Is it worthwhile to include test cases of direct types?

```
REQUIRE(ex::sender_of<my_sender0>, int)
```

In most cases where I've wanted to use this has been something like:

`template <sender_of<int> S> auto foo(S s)`

Having the extra `set_value_t` doesn't seem very pleasing or documented.

Reply:
You're right, it's not documented, which is why I haven't merged this yet. And `sender_of<int>` was how it used to work, but when P2300 got support for dependent sender types, it was no longer sufficient. Some types are only senders when used together with certain environment types, so `sender_of` needs to take an optional environment. But taking the environment last (which you would need to do if you want it to be optional) interferes with senders that send more than one value, like `sender_of< int, std::string >`. Hence, `sender_of<set_value_t(int, std::string) [, optional-env]>`.

We _could_ accept `sender_of<T [, optional-env]>` where `T` is not a function type, and interpret it as `sender_of<set_value_t(T) [, optional-env]>`. That feels a little dirty to me, but I understand its appeal. Do others have an opinion about this? @kirkshoop @griwes ?

Reply:
I would much rather have a different named concept for the single value type case. 

`sends_value<T[, Env]>`

Reply:
Thanks @ericniebler for the info.

I must admit I haven't fully absorbed the environment stuff yet and I've probably only absorbed about 40% of the rest, so take my opinion for whatever it's worth.  I'm mostly working in the coroutine model and it seems pretty handy to have some kind of concept to indicate "this type acts as if it did a `co_return T`".  That's what `sender_of<T>` did in libunifex and what seems like `sender_of<noenv, T>` does now here.  I don't have any opinion on the naming, but as a concept it seems fairly useful and a lot more consumable than with the extra `set_value_t`.  Again though, I haven't fully absorbed what it means for a sender to have multiple `set_value_t`s.

I actually arrived at this PR because I was trying to figure out why `sender_of<bool>` didn't currently work like I expected from libunifex and the P2300 examples, which lead to finding out about the environment parameter and this corresponding issue/PR.